### ISSUE TITLE
Improve this Doc Uses Wrong URL if Repo ends with ".git"

### ISSUE THREAD

**Issue by noluskey** [dotnet/docfx#2378]
**DocFX Version Used**: 2.26.0.0

**Template used**: `default`

**Steps to Reproduce**:

1. Create a repository on VSO called Repo.git (or anything that ends with `.git`)
2. Clone repository and add a simple docfx project
3. Build and visit the site, specifically looking at the "Improve This Doc" link

**Expected Behavior**:
Improve this doc should link to `https://<user>.visualstudio.com/<Project>/_git/Repo.git?<url-remainder>`

**Actual Behavior**:
Improve this doc links to `https://<user>.visualstudio.com/<Project>/_git/Repo?<url-remainder>`, which is invalid.

**Comment by noluskey** [comment]
This issue was created with the fix for issue #951 (support for submodules). However, it is valid for the root repository to contain `.git` in the name. Removing lines 140-142 in common.js:

```js
if (gitInfo.repo.substr(-4) === '.git') {
        gitInfo.repo = gitInfo.repo.substr(0, gitInfo.repo.length - 4);
}
```
would fix this, but I'm not sure the implications for the submodule issue.

**Comment by superyyrrzz** [comment]
This logic makes sense for github. For the git string `https://github.com/dotnet/docfx.git`, DocFX need to remove `.git` to construct the URL like "https://github.com/dotnet/docfx/blob/dev/xxx.md".
As vso git string doesn't contain the trailing `.git`, this logic is not needed. The fix need to move this logic into github specific place.
